packages/commonwealth/client/scripts/views/components/component_kit/new_designs/cw_text_input.tsx

```diff
@@ -0,0 +1,262 @@
+  iconRightonClick?: () => void;
+  inputValidationFn?: (value: string) => [ValidationStatus, string] | [];
+  label?: string | React.ReactNode;
+  maxLength?: number;
+  isCompact?: boolean;
+  name?: string;
+  onInput?: (e) => void;
+  onenterkey?: (e) => void;
+  onClick?: (e) => void;
+  placeholder?: string;
+  tabIndex?: number;
+  manualStatusMessage?: string;
+  manualValidationStatus?: ValidationStatus;
+};
+
+type InputStyleProps = {
+  inputClassName?: string;
+  darkMode?: boolean;
+  disabled?: boolean;
+  size?: TextInputSize;
+  width?: number | string;
+  fullWidth?: boolean;
+  validationStatus?: ValidationStatus;
+  displayOnly?: boolean;
+};
+
+type InputInternalStyleProps = {
+  hasLeftIcon?: boolean;
+  hasRightIcon?: boolean;
+  isTyping?: boolean;
+};
+
+type MessageRowProps = {
+  hasFeedback?: boolean;
+  label: string | React.ReactNode;
+  statusMessage?: string;
+  validationStatus?: ValidationStatus;
+};
+
+type TextInputProps = BaseTextInputProps &
+  InputStyleProps &
+  InputInternalStyleProps &
+  React.HTMLAttributes<HTMLDivElement>;
+
+export const MessageRow = (props: MessageRowProps) => {
+  const { hasFeedback, label, statusMessage, validationStatus } = props;
+
+  return (
+    <div
+      className={getClasses<{ hasFeedback: boolean }>(
+        { hasFeedback },
+        'MessageRow'
+      )}
+    >
+      <CWLabel label={label} />
+      {hasFeedback && (
+        <CWText
+          type="caption"
+          className={getClasses<{ status: ValidationStatus }>(
+            { status: validationStatus },
+            'feedback-message-text'
+          )}
+        >
+          {statusMessage}
+        </CWText>
+      )}
+    </div>
+  );
+};
+
+export const useTextInputWithValidation = () => {
+  const [inputTimeout, setInputTimeout] = React.useState<
+    NodeJS.Timeout | undefined
+  >();
+  const [isTyping, setIsTyping] = React.useState<boolean>(false);
+  const [statusMessage, setStatusMessage] = React.useState<
+    string | undefined
+  >();
+  const [validationStatus, setValidationStatus] = React.useState<
+    ValidationStatus | undefined
+  >();
+
+  return {
+    inputTimeout,
+    setInputTimeout,
+    isTyping,
+    setIsTyping,
+    statusMessage,
+    setStatusMessage,
+    validationStatus,
+    setValidationStatus,
+  };
+};
+
+export const CWTextInput = (props: TextInputProps) => {
+  const validationProps = useTextInputWithValidation();
+
+  const {
+    autoComplete = 'off',
+    autoFocus,
+    containerClassName,
+    darkMode,
+    defaultValue,
+    value,
+    disabled,
+    width,
+    fullWidth,
+    iconLeft,
+    iconLeftonClick,
+    iconRight,
+    iconRightonClick,
+    inputClassName,
+    inputValidationFn,
+    label,
+    maxLength,
+    name,
+    onInput,
+    onenterkey,
+    onClick,
+    placeholder,
+    isCompact = false,
+    tabIndex,
+    displayOnly,
+    manualStatusMessage = '',
+    manualValidationStatus = '',
+  } = props;
+
+  const getWidth = () => {
```

Comment:
as I mentioned couple times already, do not inject inline style => move this code to css

Reply:
I tried some solutions but apparently there is no way to edit a css class property without editing it inline. It's a easy thing to by editing it inline on the React file as the previous solution... or use some libraries like [styled-components](https://styled-components.com/).

One solution I thought was setting the default width as 240px (as per design) and have the `fullWidth` prop so the width can be set in the outer div. Example:

![image](https://github.com/hicommonwealth/commonwealth/assets/30223098/56b7bc3c-e694-4a25-ba7f-44cf213552ac)

![image](https://github.com/hicommonwealth/commonwealth/assets/30223098/703aa3b0-7ee4-4be0-ab83-6131e1fd9511)

@zakhap @meeshlin @masvelio what do you guys think?

Reply:
yeah and this is the right solution. Default is 240 but if you want to check the width of the input, you simply wrap it with the div and pass the `fullWidth` prop into the input.